src/ui/public/visualize/visualize.js

```diff
@@ -56,12 +56,8 @@ uiModules
         return Boolean(requiresSearch && isZeroHits && shouldShowMessage);
       };
 
-      $scope.showLegendTopOrLeft = function () {
-        return $scope.vis.params.legendPosition === 'top' || $scope.vis.params.legendPosition === 'left';
-      };
-
-      $scope.showLegendBottomOrRight = function () {
-        return $scope.vis.params.legendPosition === 'bottom' || $scope.vis.params.legendPosition === 'right';
+      $scope.getVisContainerClasses = function () {
+        return `vis-container--legend-${$scope.vis.params.legendPosition}`;
```

Comment:
Instead of tightly coupling to the `legendPosition` property here, we should be explicitly returning the class names we want, similar to the pattern I detail in this comment: https://github.com/elastic/kibana/pull/7781#discussion_r72876812.

Applied here, it would be something like:

```javascript
const legendPositionToVisContainerClassMap = {
  top: 'vis-container--legend-top',
  bottom: 'vis-container--legend-bottom',
  left: 'vis-container--legend-left',
  right: 'vis-container--legend-right',
};

$scope.getVisContainerClasses = function () {
  return legendPositionToVisContainerClassMap[$scope.vis.params.legendPosition];
}
```